### PR TITLE
ENG-88747 Add converters support in AI instructions

### PR DESCRIPTION
**Pull request by d-krestov**
- Introduce SchemaValidationService.ValidateConverterDeclarations which parses the SCHEMA_CONVERTERS block and flags any top-level converter key missing the required VendorPrefix.ConverterName format. The helper walks the object with a balanced parser (EnumerateTopLevelConverterKeys / TryReadConverterEntry / SkipBalancedConverterValue) so nested objects, arrays, function calls, comments, and string literals don't confuse key detection.
- Wire the new validator into PageValidateTool, PageUpdateTool, and PageSyncTool so converter format errors surface before a save and during batch sync. PageValidateTool.ContentOk now also reflects converter validity.
- Update tool descriptions so the AI agent is instructed to call get-guidance with page-schema-converters whenever the body changes SCHEMA_CONVERTERS, matching the existing pattern for handlers/validators/devkit. PageValidateTool description now explicitly states that converter keys must follow VendorPrefix.Name.
- Replace the inline "SCHEMA_CONVERTERS" literal with the new SchemaConvertersMarker constant.

Why

A converter registered without a vendor prefix (e.g. MyConverter instead of usr.MyConverter) causes a Creatio runtime error at registration time:

▎ Error when register X. Type property should have format "VendorPrefix.ConverterTypeName"

The page is saved successfully but breaks at runtime. Catching this client-side prevents the AI agent from shipping pages that fail only after deployment, and the guidance hook nudges it to read the converter contract before authoring.

Test plan

- New unit tests in SchemaValidationServiceTests.cs covering valid/invalid converter keys, empty/missing SCHEMA_CONVERTERS, nested values, comments, and edge cases.
- New E2E tests PageSyncToolE2ETests and PageValidateToolE2ETests verifying that PageSync/PageValidate surface the converter format error.

### CODE REVIEW

File: .codex/workspace-diary.md [hunk 1]
```diff
@@ -3077,3 +3077,10 @@ Decision: Restored `CreatePushPkgOptions` to read `SettingsRepository` directly
 Discovery: The breaking change came from `cd5ca8ec` / PR #549 (`feat(mcp): page operations, schema tools, DI cleanup, and IFileSystem migration`); `7baebf2e` made bootstrap intentionally use a placeholder, and #549 accidentally made `install-gate` reuse that placeholder.
 Files: C:\Projects\clio\clio\Program.cs, C:\Projects\clio\clio.tests\Command\Program.Tests.cs, C:\Projects\clio\.codex\workspace-diary.md
 Impact: The new regression test asserts that `CreateClioGatePkgOptions` leaves `Program.Container` scoped to the requested environment. Local patched `install-gate -e dev` installed cliogate successfully and `get-info -e dev` returned system info.
+
+## 2026-05-05 – ENG-88747 Added converter format validation and AI guidance hooks
+Context: Freedom UI converters registered without a `VendorPrefix.Name` key fail at runtime in Creatio with `"Error when register X. Type property should have format VendorPrefix.ConverterTypeName"`. The AI agent could ship pages that pass save and only break at runtime.
```
Comment (a-trofimchuk) [inline .codex/workspace-diary.md:3089]: do we have same validation with vendor prefix for other entries?
validator etc

**Review by a-trofimchuk — APPROVED** [review]

File: .codex/workspace-diary.md (hunk 1 above)
Comment (d-krestov) [reply .codex/workspace-diary.md:3089]: I added validation for other entries

**Review by a-trofimchuk — APPROVED** [review]
